### PR TITLE
Add some qthreads patches

### PR DESCRIPTION
**Pull request by eschnett**

### CODE REVIEW

File: var/spack/repos/builtin/packages/qthreads/restrict.patch [hunk 1]
```diff
@@ -0,0 +1,12 @@
+--- a/include/qthread/common.h.in
++++ b/include/qthread/common.h.in
+@@ -84,7 +84,9 @@
+ /* Define to the equivalent of the C99 'restrict' keyword, or to
+    nothing if this is not supported.  Do not define if restrict is
+    supported directly.  */
++#ifndef restrict
```
Comment (mathstuf) [inline var/spack/repos/builtin/packages/qthreads/restrict.patch]: `ifdef`? If it isn't defined, why `undef` it on the next line?

File: var/spack/repos/builtin/packages/qthreads/ldflags.patch [hunk 2]
```diff
@@ -0,0 +1,11 @@
+--- a/configure
++++ b/configure
+@@ -40456,7 +40456,7 @@
+   hwloc_saved_LDFLAGS="$LDFLAGS"
+   if test "x$with_hwloc" != x; then
+   CPPFLAGS="-I$with_hwloc/include $CPPFLAGS"
+-		 LDFLAGS="-L$with_hwloc/lib $CPPFLAGS"
++		 LDFLAGS="-L$with_hwloc/lib $LDFLAGS"
```
Comment (mathstuf) [inline var/spack/repos/builtin/packages/qthreads/ldflags.patch]: This looks like it should go upstream…

File: var/spack/repos/builtin/packages/qthreads/restrict.patch (hunk 1 above)
Comment (eschnett) [inline var/spack/repos/builtin/packages/qthreads/restrict.patch]: This is a `config.h.in` file for autoconf. Configure will replace the `#undef` with `#define`, depending on what it detects for the platform. If another package does the same, then `restrict` will be defined twice. Hence this (potential) `#define` needs to be protected by an `#ifdef`.

File: var/spack/repos/builtin/packages/qthreads/ldflags.patch (hunk 2 above)
Comment (eschnett) [inline var/spack/repos/builtin/packages/qthreads/ldflags.patch]: I've submitted several bug fixes upstream, but there hasn't been a release in some time.

**Comment by tgamblin** [comment]
pinged some qthreads contacts on this to see if they'll integrate it.

**Comment by npe9** [comment]
I'm working on integrating the patches today. I've applied the restrict patch and committed and closed the issue. Who should I talk to about the other two patches? @mathstuf ?

**Comment by mathstuf** [comment]
Not sure; I'm not familiar with qthreads. I'd make sure upstream is aware of them and link the issues/threads in comments from the `package.py` file.

**Pull request merged**